### DISCUSSION TITLE
Scalability (max number of documents)

### DISCUSSION THREAD

**Discussion by hartstores** [ciur/papermerge#450] [Q&A]
Hello, is there data on the scalability of papermerge ? Up to how many documents were tested ? Is there any document or guideline regarding minimum system specifications ? (CPU, RAM, etc.)

**Comment by ciur** [comment]
Not yet. You brought a good point here. Currently, my own instance features approximately a total of 4200 pages, which is roughly 600 documents (i.e. files, mostly in PDF format). To be honest Papermerge is not yet mature for real world production with millions of documents.

**Reply by aratan** [reply]
i am testing papaermerge on an aws micro with 1 g of ran and it stops every so often, i have not loaded anything yet, no files or anything, what is the minimum hardware recommendation?

**Reply by ciur** [reply]
For web app I would recommand at least 1 GB of RAM. For one worker i would recommand at least 4 GB of RAM:

https://docs.papermerge.io/3.0/setup/requirements/

In my home lab, at this moment P3 with 1 worker + PostgreSQL 16 + Solr9 fits into 8 GB of RAM (it runs on Debian12 VM). So far, my setup has aprox 4k document pages

In general it is better to count by number of pages, not documents, because one doc may have 100 pages, and there is significant cost differences between OCRing one page and 100 pages.
